### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by leifdejong**
1. Moved everything to standard Node ES6
2. Simplified package scripts
3. Added in-code documentation
4. Added additional testing (extending, disconnecting, setting, etc...)

**Note**: No provided way to extend client in browser in this release.

### CODE REVIEW

File: tests/node.js [hunk 1]
```diff
@@ -1,20 +1,93 @@
...
+          )
+        );
+
+        // kill the session
+        client.disconnect();
+      });
+  })
+  .catch(e => console.error(e));
 
+// EXTEND AND USE THE CLIENT
```
Comment (leifdejong) [inline tests/node.js]: @vkuts and @vleipnik-consensys this is how we will extend in the UI

File: package.json [hunk 2]
```diff
@@ -1,38 +1,26 @@
...
   "scripts": {
-    "start-node": "cross-env NODE_ENV=development nodemon --watch src --watch tests --exec babel-node tests/node.js",
+    "start-node": "cross-env NODE_ENV=development nodemon --watch src --watch tests --exec node tests/node.js",
     "start-browser": "cross-env NODE_ENV=development webpack-dev-server --progress --open --mode development --config webpack.config.js",
-    "build": "npm run build-browser && npm run build-node",
-    "build-node": "cross-env NODE_ENV=production npx babel src --out-dir dist/node/ --inline-source-maps",
-    "build-browser": "cross-env NODE_ENV=production webpack -p --progress --mode production --config webpack.config.js"
+    "build": "cross-env NODE_ENV=production webpack -p --progress --mode production --config webpack.config.js",
+    "watch": "cross-env NODE_ENV=development nodemon --watch src --watch tests --exec npm run build",
+    "publish-locally": "npm link",
```
Comment (leifdejong) [inline package.json]: @vkuts this allows us to test it as an npm package locally before officially releasing

File: src/core.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+      if (typeof params.key !== 'string') {
+        if (typeof params.key.kid === 'undefined' || typeof params.key.private === 'undefined' ) {
+          throw new Error(`Invalid Params: ${this._pkg.name} RSA auth requires a "key.kid" and a "key.private".`)
+        }
+      }
+
+      // properties
+      this._key = params.key;
+      this._protocol = params.protocol || `https`
+      this._host = params.host || 'consensys-mesh.ga';
```
Comment (vkuts) [inline src/core.js]: Might have to make this `consensys-mesh.com` in light of our recent DNS issue

File: src/core.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+      this._namespace = params.namespace || 'api';
+      this._version = params.version || 'v1';
+      this._endpoint = params.endpoint || `${this._protocol}://${this._host}/${this._namespace}/${this._version}`;
+
+      // setters
+      this.setOrg(params.orgId);
```
Comment (vkuts) [inline src/core.js]: orgId seems like a required param, even though there's a setter for it to change it later. Might want to enforce it as ctor input.

Comment (leifdejong) [reply src/core.js]: @vkuts  It's not a required param. It is only required if you are a member of multiple organizations.

File: src/core.js (hunk 3 above)
Comment (leifdejong) [reply src/core.js]: Wilco

File: src/core.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        this._session = session;
+
+        // handle orgIds
+        // - if there is one, use it
+        // - if there are multiple, use this._orgId
+        if (session.orgIds.length === 1) {
+          this.setOrg(session.orgIds[0]);
+        } else {
+          if (!session.orgIds.includes(this._orgId)) {
+            throw new Error(`Invalid Param: ${this._pkg.name} requires a valid "orgId".`)
```
Comment (leifdejong) [inline src/core.js]: @vkuts

File: package.json (hunk 2 above)
Comment (vkuts) [reply package.json]: That's def nice 👍

**Pull request merged**